### ISSUE TITLE
Spinner accidentally submits form

### ISSUE THREAD

**Issue by m00k** [primefaces/primeng#2079]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

http://plnkr.co/edit/o24mQt3h1K8F63w4XV7w

**Current behavior**

Place a spinner control inside a (reactive) form. Clicking the up/down buttons of the spinner accidentally causes form submits.

**Expected behavior**

Clicking the up/down buttons of a spinner control should NOT submit.

**Minimal reproduction of the problem with instructions**

In the provided plunker when trying to use the spinner it will submit the form which is undesired. I think https://github.com/primefaces/primeng/commit/3c96786005675a65f684930f5aab897d81a22b74 is the change that causes the issue. The type attributes of the spinner buttons need to be set to type="button". If not set they default to "submit" as described in https://angular.io/docs/ts/latest/guide/reactive-forms.html.

* **Angular version:** 2.4.X

* **PrimeNG version:** 2.0.0

**Comment by kuba-010** [comment]
Came across this bug today - Probably worth pointing out this happens in Edge and but not in Chrome. Haven't tested any other browsers but definitely a problem!

**Comment by m00k** [comment]
@kuba-010  that is NOT correct. issue can be observed also with **Chrome** (55.0), **Chromium** (55.0), **Firefox** (51.0.1) (tested on a Ubuntu Gnome 16.10 system).

**Comment by ducmaxime** [comment]
Same here with firefox and chrome !!

Any Solutions ?

**Comment by kuba-010** [comment]
@m00k - checked again against the plunk, you're right it just seems to behave a little different in my app in chrome. Anyway definitely needs fixing, only mentioned it in case other people weren't seeing the issue

**Comment by stephanie-dm** [comment]
I ran into this problem as well today in Chrome. I suppose there's no workaround?

**Comment by ducmaxime** [comment]
Yes, in my case I did change the code of nodes_modules/primeng/components/spinner/spinner.js like this

     core_1.Component({
            selector: 'p-spinner',
            template: "\n        <span class=\"ui-spinner ui-widget ui-corner-all\">\n            <input #in pInputText type=\"text\" class=\"ui-spinner-input\" [value]=\"valueAsString\"\n            [attr.size]=\"size\" [attr.maxlength]=\"maxlength\" [attr.tabindex]=\"tabindex\" [attr.placeholder]=\"placeholder\" [disabled]=\"disabled\" [readonly]=\"readonly\"\n            (keydown)=\"onInputKeydown($event)\" (keyup)=\"onInput($event,in.value)\" (keypress)=\"onInputKeyPress($event)\" (blur)=\"onBlur()\" (change)=\"handleChange($event)\" (focus)=\"onFocus()\">\n            <button type=\"button\" [ngClass]=\"{'ui-spinner-button ui-spinner-up ui-corner-tr ui-button ui-widget ui-state-default':true,'ui-state-disabled':disabled}\" [disabled]=\"disabled\"\n                (mouseleave)=\"onUpButtonMouseleave($event)\" (mousedown)=\"onUpButtonMousedown($event,in)\" (mouseup)=\"onUpButtonMouseup($event)\">\n                <span class=\"fa fa-caret-up\"></span>\n            </button>\n            <button type=\"button\" [ngClass]=\"{'ui-spinner-button ui-spinner-down ui-corner-br ui-button ui-widget ui-state-default':true,'ui-state-disabled':disabled}\" [disabled]=\"disabled\"\n                (mouseleave)=\"onDownButtonMouseleave($event)\" (mousedown)=\"onDownButtonMousedown($event,in)\" (mouseup)=\"onDownButtonMouseup($event)\">\n                <span class=\"fa fa-caret-down\"></span>\n            </button>\n        </span>\n    ",
            host: {
                '[class.ui-inputwrapper-filled]': 'filled',
                '[class.ui-inputwrapper-focus]': 'focus'
            },
            providers: [domhandler_1.DomHandler, exports.SPINNER_VALUE_ACCESSOR],
        }),

**Comment by m00k** [comment]
looks like this was fixed for v2.0.1 with https://github.com/primefaces/primeng/commit/2b4333321433656ca6eaec99bf4da8c1345974a7
